CMakeLists.txt

```diff
@@ -37,3 +47,23 @@ add_subdirectory(src/go)
 add_subdirectory(src/c)
 
 install(DIRECTORY "pixmaps" DESTINATION "${PURPLE_DATA_DIR}" FILES_MATCHING PATTERN "*.png")
+
+# package settings
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Whatsapp protocol plug-in for libpurple with pidgin")
+set(CPACK_PACKAGE_VENDOR "Bryant Eadon")
+set(CPACK_PACKAGE_DESCRIPTION "whatsmeow is a go-based whatsapp protocol plug-in for libpurple based on the whatsapp business apis.")
+set(CPACK_PACKAGE_CONTACT "bryant@balancebitsconsulting.com")
+set(CPACK_PACKAGE_VERSION_MAJOR "${VERSION_MAJOR}")
+set(CPACK_PACKAGE_VERSION_MINOR "${VERSION_MINOR}")
+set(CPACK_PACKAGE_VERSION_PATCH "${VERSION_PATCH}")
+set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+set(CPACK_DEBIAN_PACKAGE_ARCHITECTURE "${ARCHITECTURE}")
+set(CPACK_SOURCE_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}_${VERSION}_${CPACK_DEBIAN_PACKAGE_ARCHITECTURE}")
+set(CPACK_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}_${VERSION}_${CPACK_DEBIAN_PACKAGE_ARCHITECTURE}")
+set(CPACK_DEBIAN_PACKAGE_DEPENDS "libpurple0 (>= 2.10.0), libglib2.0-0 (>= 2.24), libjson-glib-1.0-0 (>= 0.8.0)")
+set(CPACK_DEBIAN_PACKAGE_MAINTAINER "Bryant Eadon") #required
+
+set(PACK "DEB" CACHE STRING "Generate a Package")
+set(CPACK_GENERATOR ${PACK})

```

Comment:
Is there a reason we need `${PACK}` and not write `${CPACK_GENERATOR}` directly?

Reply:
Not specifically -- It was the code I was following from the skype4pidgin plugin.  I am sure there will be a reason to break it up into 2 lines at some later point as we build other packages, but for now I agree it is overly verbose.

I have not attempted to pack by combining the lines, but I imagine it would work just fine.  Would you like to see the 2 lines collapsed ?

Reply:
Yes, I prefer collapsed lines (done).

Reply:
k, resolved.